File: 3797-design-spreadsheet/design-spreadsheet.py
class Spreadsheet:

    def __init__(self, rows: int):
        # Number of rows in the spreadsheet (not really used directly,
        # but could help validate input if needed)
        self.rows = rows

        # Dictionary to store cell values
        # Key = cell reference (e.g., "A1"), Value = integer
        self.cells = {}

    def setCell(self, cell: str, value: int) -> None:
        # Store/update the given cell with the specified value
        self.cells[cell] = value

    def resetCell(self, cell: str) -> None:
        # Reset the cell’s value to 0
        self.cells[cell] = 0

    def val(self, x):
        # Helper function to evaluate a token (either a cell reference or an integer)
        if x[0].isalpha():  # If the token starts with a letter, it's a cell reference
            return self.cells.get(x, 0)  # Return stored value, default to 0 if unset
        else:
            return int(x)  # Otherwise, it's just a plain number

    def getValue(self, formula: str) -> int:
        # Formula format is always "=X+Y"
        a, b = formula[1:].split("+")  # Remove '=' then split at '+'
        return self.val(a) + self.val(b)  # Evaluate both parts and sum

# Your Spreadsheet object will be instantiated and called as such:
# obj = Spreadsheet(rows)
# obj.setCell(cell,value)
# obj.resetCell(cell)
# param_3 = obj.getValue(formula)

File: 3797-design-spreadsheet/design-spreadsheet.java
class Spreadsheet {
    private int rows;
    private Map<String, Integer> cells;

    public Spreadsheet(int rows) {
        // Initialize spreadsheet with given rows
        this.rows = rows;
        this.cells = new HashMap<>();
    }

    public void setCell(String cell, int value) {
        // Set/update the value of the given cell
        cells.put(cell, value);
    }

    public void resetCell(String cell) {
        // Reset the cell to 0
        cells.put(cell, 0);
    }

    // Helper method to evaluate either a number or a cell reference
    private int val(String x) {
        if (Character.isLetter(x.charAt(0))) {
            // It's a cell reference
            return cells.getOrDefault(x, 0);
        } else {
            // It's a plain number
            return Integer.parseInt(x);
        }
    }

    public int getValue(String formula) {
        // Formula is of the form "=X+Y"
        String expr = formula.substring(1); // remove "="
        String[] parts = expr.split("\\+"); // split at "+"
        return val(parts[0]) + val(parts[1]);
    }
}

/**
 * Usage:
 * Spreadsheet obj = new Spreadsheet(10);
 * obj.setCell("A1", 5);
 * obj.setCell("B1", 3);
 * System.out.println(obj.getValue("=A1+B1")); // 8
 */
